templates/stories/page-switch.html

```diff
@@ -1,22 +1,72 @@
 <script>
+  var COOKIE_AMP_EXP = 'AMP_EXP';
+var AMP_STORY = 'amp-story';
+
+// Move story to the selected page
+var pageId = getUrlParameter('id');
+switchToPage(pageId);
+
+function switchToPage(pageId) {
+  if (!pageId) {
+    return;
+  }
+  var story = document.querySelector('amp-story');
+  story.addEventListener('ampstory:load', function() {
+    var event = new CustomEvent('ampstory:switchpage',  { 
+      bubbles: true, 
+      detail: { 
+        targetPageId: pageId
+      }
+    });
+    story.dispatchEvent(event);
+  });
+}
+
 function getUrlParameter(name) {
-    var url = window.location.href;
-    name = name.replace(/[\[\]]/g, "\\$&");
-    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
-        results = regex.exec(url);
-    if (!results) return null;
-    if (!results[2]) return '';
-    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  var url = window.location.href;
+  name = name.replace(/[\[\]]/g, "\\$&");
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+    results = regex.exec(url);
+  if (!results) return null;
+  if (!results[2]) return '';
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
 }
-const story = document.querySelector('amp-story');
-const pageId = getUrlParameter('id');
-story.addEventListener('ampstory:load', function() {
-  const event = new CustomEvent('ampstory:switchpage',  { 
-    bubbles: true, 
-    detail: { 
-      targetPageId: pageId
+
+// Enable story experiment
+enableExperiment();
+
+function enableExperiment() {
+  var expCookie = getCookie(COOKIE_AMP_EXP);
+  if (expCookie) {
+    if (expCookie.indexOf(AMP_STORY) != -1) {
+      return;
+    } else {
+      expCookie += '%2C' + AMP_STORY;
```

Comment:
Scattering some `encodeURIComponent()` and `decodeURIComponent()` through the `setCookie()` and `getCookie()` functions might eliminate the need for `%2C`. Or replace `%2C` with `encodeURIComponent(' ')`. Depends how "helpful" the helper functions are supposed to be I guess.

Reply:
they're only used in a single place, so I didn't bother. But you're right - fixed it.